Google.Cast/source/Google.Cast/Extensions.cs

```diff
@@ -122,57 +122,73 @@ public partial class Logger
 		}
 
 
-		public void Log (string function, params string[] message)
+		public void Log (string function, string format, params object [] args)
 		{
-			if (message == null)
-				throw new ArgumentNullException ("Message");
+			var message = string.Format (format, args);
+			_Log (function, message);
+		}
+	}
 
-			var pNativeArr = Marshal.AllocHGlobal ((message.Length - 1) * IntPtr.Size);
+	public class LoggerHandler : NSObject, ILoggerDelegate
```

Comment:
How about this instead?  


	public class LoggerHandlerBase : NSObject, ILoggerDelegate
	{
		public abstract void Log (string function, string message);

		public void _Log (IntPtr function, string message)
		{
			var func = Marshal.PtrToStringAnsi (function);

			Log (func, message);
		}
	}